plugin/session_buffer.py

```diff
@@ -311,9 +311,18 @@ def on_text_changed_async(self, view: sublime.View, change_count: int,
                 self._pending_changes.update(change_count, changes)
                 purge = True
             if purge:
+                self._cancel_pending_requests_async()
                 debounced(lambda: self.purge_changes_async(view), FEATURES_TIMEOUT,
                           lambda: view.is_valid() and change_count == view.change_count(), async_thread=True)
 
+    def _cancel_pending_requests_async(self) -> None:
+        if self._document_diagnostic_pending_request:
+            self.session.cancel_request(self._document_diagnostic_pending_request.request_id)
+            self._document_diagnostic_pending_request = None
+        if self.semantic_tokens.pending_response:
+            self.session.cancel_request(self.semantic_tokens.pending_response)
```

Comment:
I have an urge to rename `pending_response` to `pending_request`.

`pending_response` sounds like a boolean state (waiting for response) rather than what it actually is.

Reply:
My thinking was that the request has already happened (was sent), so we're indeed only still waiting for the response now. But maybe we use the word "pending" in a sligthly different meaning. I guess it can both mean something like "upcoming" (that's how I used it here), or "unanswered". Of course the server response is directly coupled to the request, so feel free to rename it if you like to.